sapl/base/views.py

```diff
@@ -915,6 +919,339 @@ def get(self, request, *args, **kwargs):
+            lista_bancada_autor_externo.append(
+                (q_autor_externo, bancada, 'Bancada', 'sistema/bancada')
+            )
+
+    for comissao in Comissao.objects.all().order_by('nome'):
+        autor_externo = comissao.autor.filter(tipo=tipo_autor_externo)
+
+        if len(autor_externo):
+            q_autor_externo = comissao.autor.get(tipo=tipo_autor_externo)
+            lista_comissao_autor_externo.append(
+                (q_autor_externo, comissao, 'Comissão', 'comissao')
+            )
+
+    return lista_bancada_autor_externo + lista_comissao_autor_externo
+
+
+class ListarBancadaComissaoAutorExternoView(PermissionRequiredMixin, ListView):
+    model = get_user_model()
+    template_name = 'base/bancada_comissao_autor_externo.html'
+    context_object_name = 'bancada_comissao_autor_externo'
+    permission_required = ('base.list_appconfig',)
+    paginate_by = 10
+
+    def get_queryset(self):
+        return bancada_comissao_autor_externo()
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            ListarBancadaComissaoAutorExternoView, self
+            ).get_context_data(**kwargs)
+        paginator = context['paginator']
+        page_obj = context['page_obj']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        context[
+            'NO_ENTRIES_MSG'
+            ] = 'Nenhum encontrado.'
+        return context
+
+
+def autores_duplicados():
+    lista_duplicados = []
+    autores_duplicados = Autor.objects.values(
+        'nome', 'tipo__descricao').annotate(
+            count=Count('nome')).filter(count__gt=1)
+
+    for autor in autores_duplicados:
+        lista_duplicados.append(autor.values())
+
+    return lista_duplicados
+
+
+class ListarAutoresDuplicadosView(PermissionRequiredMixin, ListView):
+    model = get_user_model()
+    template_name = 'base/autores_duplicados.html'
+    context_object_name = 'autores_duplicados'
+    permission_required = ('base.list_appconfig',)
+    paginate_by = 10
+
+    def get_queryset(self):
+        return autores_duplicados()
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            ListarAutoresDuplicadosView, self).get_context_data(**kwargs)
+        paginator = context['paginator']
+        page_obj = context['page_obj']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        context[
+            'NO_ENTRIES_MSG'
+            ] = 'Nenhum encontrado.'
+        return context
+
+
+def parlamentares_mandatos_intersecao():
+    intersecoes = []
+
+    for parlamentar in Parlamentar.objects.all().order_by('nome_completo'):
+        mandatos = parlamentar.mandato_set.all()
+        combinacoes = itertools.combinations(mandatos, 2)
+
+        for c in combinacoes:
+            if c[0].data_inicio_mandato and c[1].data_inicio_mandato:
+                if c[0].data_fim_mandato and c[1].data_fim_mandato:
```

Comment:
Se não tiver `data_fim_mandato` poderia colocar como `timezone.now().datetime`? 

Eu colocaria data_inicio e data_fim em variáveis antes de passar para a função que testa as interseções.

Reply:
Sim. Alterei a função:
```suggestion
def parlamentares_mandatos_intersecao():
    intersecoes = []

    for parlamentar in Parlamentar.objects.all().order_by('nome_completo'):
        mandatos = parlamentar.mandato_set.all()
        combinacoes = itertools.combinations(mandatos, 2)

        for c in combinacoes:
            data_inicio_mandato1 = c[0].data_inicio_mandato
            data_fim_mandato1 = c[0].data_fim_mandato if c[0].data_fim_mandato else timezone.now().date()

            data_inicio_mandato2 = c[1].data_inicio_mandato
            data_fim_mandato2 = c[1].data_fim_mandato if c[1].data_fim_mandato else timezone.now().date()

            if data_inicio_mandato1 and data_inicio_mandato2:
                exists = intervalos_tem_intersecao(
                    data_inicio_mandato1, data_fim_mandato1,
                    data_inicio_mandato2, data_fim_mandato2)
                if exists:
                    intersecoes.append((parlamentar, c[0], c[1]))

    return intersecoes                
```